src/DIRAC/FrameworkSystem/DB/TokenDB.py

```diff
@@ -87,35 +87,49 @@ def getTokenForUserProvider(self, userID, provider):
             return self.__result(session, S_ERROR(str(e)))
         return self.__result(session, S_OK(OAuth2Token(self.__rowToDict(token)) if token else None))
 
-    def updateToken(self, token, userID, provider, rt_expired_in):
-        """Update tokens
+    def updateToken(self, token: dict, userID: str, provider: str, rt_expired_in: int):
+        """Store or update an existing token in the database.
+        Before saving, the token is checked for expired.
+        Also, the database cannot contain several user tokens signed by one provider,
+        only one with the maximum possible permissions is enough.
 
-        :param dict token: token info
-        :param str userID: user ID
-        :param str provider: provider
-        :param int rt_expired_in: refresh token lifetime
+        :param token: token information dictionary
+        :param userID: user ID (token owner)
+        :param provider: provider name that issued the token
+        :param rt_expired_in: refresh token expiration time, will be applied if the rt_expires_at value is missing
```

Comment:
You could have kept the type? (e.g. `:param str userID:`)

Reply:
it is in the annotation of the method, should this be enough?

Reply:
Ah right sorry.